### PR TITLE
pcem: init at 17

### PR DESCRIPTION
**Pull request by terinjokes**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8879,6 +8879,12 @@
     githubId = 649832;
     name = "Teo Klestrup Röijezon";
   };
+  terin = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8882]: Please put this into a separate commit with the message ``maintainers: add terin``.

File: pkgs/misc/emulators/pcem/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchzip, wxGTK31, coreutils, SDL2, openal, alsaLib, pkg-config
+, autoreconfHook, withNetworking ? true, withALSA ? true, ... }:
+
+let inherit (stdenv.lib) optional;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:4]:
```suggestion
```

File: pkgs/misc/emulators/pcem/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+in stdenv.mkDerivation rec {
+  pname = "pcem";
+  version = "17";
+
+  buildInputs = [ wxGTK31 coreutils SDL2 openal ] ++ optional withALSA alsaLib;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:10]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
  buildInputs = [ wxGTK31 coreutils SDL2 openal ] ++ lib.optionals withALSA [ alsaLib ];
```

File: pkgs/misc/emulators/pcem/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchzip, wxGTK31, coreutils, SDL2, openal, alsaLib, pkg-config
+, autoreconfHook, withNetworking ? true, withALSA ? true, ... }:
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:2]:
```suggestion
{ stdenv, fetchzip, wxGTK31, coreutils, SDL2, openal, alsaLib, pkg-config
, autoreconfHook, withALSA ? true }:
```

File: pkgs/misc/emulators/pcem/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  version = "17";
+
+  buildInputs = [ wxGTK31 coreutils SDL2 openal ] ++ optional withALSA alsaLib;
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+
+  src = fetchzip {
+    url = "https://pcem-emulator.co.uk/files/PCemV${version}Linux.tar.gz";
+    stripRoot = false;
+    sha256 = "067pbnc15h6a4pnnym82klr1w8qwfm6p0pkx93gx06wvwqsxvbdv";
+  };
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:17]:
```suggestion
```

File: pkgs/misc/emulators/pcem/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchzip, wxGTK31, coreutils, SDL2, openal, alsaLib, pkg-config
+, autoreconfHook, withNetworking ? true, withALSA ? true, ... }:
+
+let inherit (stdenv.lib) optional;
+
+in stdenv.mkDerivation rec {
+  pname = "pcem";
+  version = "17";
+
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:9]:
```suggestion
  version = "17";

  src = fetchzip {
    url = "https://pcem-emulator.co.uk/files/PCemV${version}Linux.tar.gz";
    stripRoot = false;
    sha256 = "067pbnc15h6a4pnnym82klr1w8qwfm6p0pkx93gx06wvwqsxvbdv";
  };
```

File: pkgs/misc/emulators/pcem/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  buildInputs = [ wxGTK31 coreutils SDL2 openal ] ++ optional withALSA alsaLib;
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:11]:
```suggestion
```

File: pkgs/misc/emulators/pcem/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchzip {
+    url = "https://pcem-emulator.co.uk/files/PCemV${version}Linux.tar.gz";
+    stripRoot = false;
+    sha256 = "067pbnc15h6a4pnnym82klr1w8qwfm6p0pkx93gx06wvwqsxvbdv";
+  };
+
+  configureFlags = [ "--enable-release-build" ]
+    ++ optional withNetworking "--enable-networking"
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:20]:
```suggestion
  configureFlags = [ "--enable-release-build" "--enable-networking" ]
```

Not sure why we want this flag.

File: pkgs/misc/emulators/pcem/default.nix [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    ++ optional withALSA "--enable-alsa";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/pcem/default.nix:21]:
```suggestion
    ++ lib.optionals withALSA [ "--enable-alsa" ];
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please change the PR title and commit title to ``pcem: init at 17``.

File: pkgs/top-level/all-packages.nix [hunk 10]
```diff
@@ -28560,6 +28560,8 @@ in
 
   OSCAR = qt5.callPackage ../applications/misc/OSCAR { };
 
+  pcem = callPackage ../misc/emulators/pcem {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:28563]:
```suggestion
  pcem = callPackage ../misc/emulators/pcem { };
```

File: pkgs/misc/emulators/pcem/default.nix (hunk 8 above)
Comment (terinjokes) [reply pkgs/misc/emulators/pcem/default.nix:20]: This allows you to disable emulation of any network cards and the in-tree libslirp build. I know I wouldn't want to put any of these VMs on my network accidentally, and debated if this should be default true or default false.

File: pkgs/misc/emulators/pcem/default.nix (hunk 9 above)
Comment (terinjokes) [reply pkgs/misc/emulators/pcem/default.nix:21]: Since I'm only adding one element to the list, I prefer `optional` over `optionals`. I see it's widely used, and I don't see any documentation that says not to use it.

File: pkgs/misc/emulators/pcem/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/misc/emulators/pcem/default.nix:20]: This should be a runtime setting but fine with me.

File: pkgs/misc/emulators/pcem/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/misc/emulators/pcem/default.nix:4]: This shouldn't be done here.

File: pkgs/misc/emulators/pcem/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/misc/emulators/pcem/default.nix:2]: The ... should be removed.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please change the commit title to ``pcem: init at 17``
